src/CookiePhysicsComponent.cpp

```diff
@@ -73,11 +74,45 @@ void CookiePhysicsComponent::updatePhysics(float deltaTime) {
         // The color change has to be done here,
         // due to the bouncing off, the cookie never really hits the object.
         // TODO(nurgan) implement checking for object type and only change color if it is a "house"
-        if(objHit->getRenderComponent()->getShader().compare("Green")) {
+        if( objHit->getRenderComponent()->getShader().compare("Green")) {
             objHit->changeShader("Green");
+            cookieState.hits++;
+            cookieState.hitPosition = holder_->getPosition();
+            float score = calculateScore();
+            GameManager& gameManager = GameManager::instance();
+            gameManager.reportScore(score);
+            gameManager.increaseTime(score/100.0);
         }
 
     }
 
 
 }
+
+float CookiePhysicsComponent::calculateScore() {
+
+    //first house hit 500, second 1000, ...
+    float score = 500.0 * cookieState.hits;
+
+    //the distance multiplier
+    float distance = glm::distance(
+            glm::vec3(cookieState.launchPosition.x, 0.0, cookieState.launchPosition.z),
+            glm::vec3(cookieState.hitPosition.x, 0.0, cookieState.hitPosition.z) );
```

Comment:
Again, which distance does this apply to? The cookies final position or the first hit position?